### PR TITLE
Additional TooltipTrigger Tests

### PR DESCRIPTION
**Pull request by MidnightCoder06**
Closes https://jira.corp.adobe.com/browse/RSP-1512

### CODE REVIEW

**Comment by MidnightCoder06** [comment]
Still not sure why the closing actions aren't being registered in tooltiptrigger tests file 🤔 so I deleted them instead of doing the id checking hack. Technically this is handled with the tooltip manager tests file anyway 🤷‍♂  
For some reason pointer events are not being registered for me in the testing framework either but clearly can be / are being used. UsePress.test seems to be doing some hacky things to get them to be registered.
I'm thinking we merge what is here and leave that to be explored later.

File: packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js [hunk 1]
```diff
@@ -34,10 +34,6 @@ describe('TooltipTrigger', function () {
         </Provider>
       );
 
-      expect(() => {
```
Comment (snowystinger) [inline packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js]: why was this removed?

File: packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js [hunk 2]
```diff
@@ -55,7 +51,7 @@ describe('TooltipTrigger', function () {
 
   describe('click related tests', function () {
 
-    it('triggered by click event', async function () {
+    it('a click event can open the tooltip', async function () {
```
Comment (snowystinger) [inline packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js]: what's the difference between this test and the one above it?

File: packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js [hunk 3]
```diff
@@ -65,22 +61,17 @@ describe('TooltipTrigger', function () {
...
       let tooltip = getByRole('tooltip');
 
-      // wait for appearance
       await wait(() => {
         expect(tooltip).toBeVisible();
       });
     });
 
-    it('pressing esc should close the tooltip after a click event', async function () {
+    it('pressing escape should close the tooltip after a click event', async function () {
```
Comment (snowystinger) [inline packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js]: why does this one check toBeInTheDocument instead of toBeVisible?

Comment (snowystinger) [reply packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js]: why does this one use await tooltip toBeInTheDocument && await waitForDomChange? why don't any of the others use waitForDomChange or why isn't waitForDomChange replaced with await tooltip not toBeInTheDocument?

File: packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js [hunk 4]
```diff
@@ -106,4 +122,80 @@ describe('TooltipTrigger', function () {
...
+            <ActionButton>Trigger</ActionButton>
+            <Tooltip>content</Tooltip>
+          </TooltipTrigger>
+        </Provider>
+      );
+
+      let button = getByText('Trigger');
+      fireEvent.mouseOver(button);
+
+      await new Promise((b) => setTimeout(b, 300));
```
[the remark this answers is not in the document]
Comment (snowystinger) [reply packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js]: Any instance of setTimeout should also be paired with jest timer mocking https://jestjs.io/docs/en/timer-mocks.html `jest.useFakeTimers()`

If you're using it to wait for the dom to change and there are no setTimeouts involved, please use those functions instead

File: packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js [hunk 5]
```diff
@@ -106,4 +97,54 @@ describe('TooltipTrigger', function () {
...
+            <ActionButton>Trigger</ActionButton>
+            <Tooltip>content</Tooltip>
+          </TooltipTrigger>
+        </Provider>
+      );
+
+      let button = getByText('Trigger');
+      fireEvent.mouseOver(button);
+
+      await new Promise((c) => setTimeout(c, 300));
```
Comment (snowystinger) [inline packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js]: same thing

File: packages/@react-stately/tooltip/test/TooltipManager.test.js [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+    let tooltip2 = {open: false, setOpen: setOpenSpy2, tooltipManager};
+    let triggerId2 = 'triggerId-2';
+
+    tooltipManager.showTooltipDelayed(tooltip2, triggerId2);
+
+    expect(tooltipManager.hoverShowTimeout).not.toBeNull();
+    expect(tooltipManager.hoverHideTimeout).toBeNull();
+
+    // run past first tooltips show timer
+    jest.advanceTimersByTime(150);
```
Comment (snowystinger) [inline packages/@react-stately/tooltip/test/TooltipManager.test.js]: i think there should be a check for something right after this line?
originally I'd checked 
    `expect(tooltipManager.visibleTooltips).toBeNull();`
is there a reason for not checking this?

File: packages/@react-stately/tooltip/test/TooltipManager.test.js [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+    // finish all the timers
+    jest.runAllTimers();
```
Comment (snowystinger) [inline packages/@react-stately/tooltip/test/TooltipManager.test.js]: I had originally outlined checking
`expect(setOpenSpy).not.toHaveBeenCalled();`
is there a reason for not checking this?

File: packages/@react-stately/tooltip/test/TooltipManager.test.js [hunk 8]
```diff
@@ -0,0 +1,137 @@
+import {TooltipManager} from '../';
+
+describe('TooltipManager', () => {
```
Comment (snowystinger) [inline packages/@react-stately/tooltip/test/TooltipManager.test.js]: I'm assuming that the remainder of the tests will be added in https://jira.corp.adobe.com/browse/RSP-1471
Can you please make notes of places that you diverged from the tests outlined there vs here and why?

File: packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js (hunk 1 above)
Comment (snowystinger) [reply packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js]: is this because of this comment?
> Still not sure why the closing actions aren't being registered in tooltiptrigger tests file 🤔 so I deleted them instead of doing the id checking hack. Technically this is handled with the tooltip manager tests file anyway 🤷‍♂

Lets try to figure out why, I have some time today, maybe we can look at it together after lunch?

File: packages/@react-aria/interactions/test/DOMPropsResponder.test.js [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+  it('should handle hover events on nested hoverable children', function () {
+    let onHover = jest.fn();
+    let {getByRole} = render(
+      <DOMPropsResponder>
+        <div>
+          <Hoverable onHover={onHover}>
+            <button>Button</button>
+          </Hoverable>
+        </div>
+      </DOMPropsResponder>
```
Comment (LFDanLu) [inline packages/@react-aria/interactions/test/DOMPropsResponder.test.js]: Ah I think I understand what you are trying to test here now, these are like the PressResponder tests yeah? You should probably move the `onHover` and `ref` props to `DOMPropsResponder` so it actually tests that the `DOMPropsResponder` context is being set up properly and being consumed by `Hoverable`.

```
<DOMPropsResponder onHover={onHover}>
....
```

File: packages/@react-stately/tooltip/test/TooltipManager.test.js (hunk 8 above)
Comment (MidnightCoder06) [reply packages/@react-stately/tooltip/test/TooltipManager.test.js]: That's a toast container ticket? 
I'll look at your outline again

File: packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js (hunk 1 above)
Comment (MidnightCoder06) [reply packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js]: No I just didn't think that was a necessary part of that test. I'll add it back

Comment (MidnightCoder06) [reply packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js]: O actually that did cause a failure... not sure why

File: packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js (hunk 2 above)
Comment (MidnightCoder06) [reply packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js]: In the first one I am checking Id's (line 59 and and 60). In the one you referenced here I am not

File: packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js (hunk 3 above)
Comment (MidnightCoder06) [reply packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js]: I wasn't sure which was the best to use. 
I pushed a change to use inDocument / notInDocument instead any visibility checks.
Do you have a preference for the waitForDomChange, etc ?

File: packages/@react-stately/tooltip/test/TooltipManager.test.js (hunk 7 above)
Comment (MidnightCoder06) [reply packages/@react-stately/tooltip/test/TooltipManager.test.js]: I do expect this to have been called. The delay stops it's appearance but the method is still called

File: packages/@react-stately/tooltip/test/TooltipManager.test.js (hunk 6 above)
Comment (MidnightCoder06) [reply packages/@react-stately/tooltip/test/TooltipManager.test.js]: added 👍

File: packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js (hunk 2 above)
Comment (snowystinger) [reply packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js]: Right, so why do we need this test, what is the value that this test adds over the previous one?

File: packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js (hunk 3 above)
Comment (snowystinger) [reply packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js]: I'm resolving this comment for now given the work we did yesterday to try and switch to fake timers. Once we have that in place, we should be able to sort this. But that'll be a later PR

File: packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (snowystinger) [reply packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js]: same thing, resolving for later PR based on our fake timers work yesterday

File: packages/@react-stately/tooltip/test/TooltipManager.test.js (hunk 8 above)
Comment (snowystinger) [reply packages/@react-stately/tooltip/test/TooltipManager.test.js]: yup, whoops, copy pasta mistake https://jira.corp.adobe.com/browse/RSP-1367

File: packages/@react-stately/tooltip/test/TooltipManager.test.js (hunk 6 above)
Comment (snowystinger) [reply packages/@react-stately/tooltip/test/TooltipManager.test.js]: I don't see the line added?

File: packages/@react-stately/tooltip/test/TooltipManager.test.js (hunk 7 above)
Comment (snowystinger) [reply packages/@react-stately/tooltip/test/TooltipManager.test.js]: ah, had a slightly different expectation based off the implementation i proposed, this is fine
